src/main.c

```diff
@@ -35,36 +43,49 @@ void main(void) {
         printk("init storage failed (err %d)\n", err);
         return;
     }
+	#endif
 
     err = init_io();
     if (err) {
         printk("Button init failed (err %d)\n", err);
         return;
     }
 
-    /* Initialize the Bluetooth Subsystem */
-    err = bt_enable(NULL);
-    if (err) {
-        printk("Bluetooth init failed (err %d)\n", err);
-        return;
-    }
+	#if BLUETOOTH
+	/* Initialize the Bluetooth Subsystem */
+	err = bt_enable(NULL);
+	if (err) {
+		printk("Bluetooth init failed (err %d)\n", err);
+		return;
+	}
 
     printk("Bluetooth initialized\n");
 
-    err = init_gatt();
-    if (err) {
-        printk("init gatt failed(err %d)\n", err);
-        return;
-    }
+	err = init_gatt();
+	if (err) {
+		printk("init gatt failed(err %d)\n", err);
+		return;
+	}
+	#endif
 
     err = init_covid();
     if (err) {
         printk("init covid failed (err %d)\n", err);
         return;
     }
 
-    do {
-        do_covid();
-        do_gatt();
-    } while (1);
+	printk("init display\n");
+	err = init_display();
+	if (err) {
+		printk("init display failed (err %d)\n", err);
+	}
+    
+    struct k_thread display_thread_data;
+    k_tid_t display_tid = k_thread_create(&display_thread_data, display_stack_area, K_THREAD_STACK_SIZEOF(display_stack_area), display_thread, NULL, NULL, NULL, 0, 0, K_NO_WAIT);
```

Comment:
Please put this code into the init_display method so the main function stays clean :)